projects/components/src/table/data/table-cdk-column-util.ts

```diff
@@ -8,7 +8,12 @@ export namespace TableCdkColumnUtil {
   ): changedColumn is TableColumnConfig => changedColumn !== undefined;
 
   export const removeNonDataConfigs = (columnConfigs: TableColumnConfig[]): TableColumnConfig[] =>
-    columnConfigs.filter(columnConfig => columnConfig.renderer !== StandardTableCellRendererType.RowExpander);
+    columnConfigs.filter(
+      columnConfig =>
+        columnConfig.renderer !== StandardTableCellRendererType.RowExpander &&
+        columnConfig.renderer !== StandardTableCellRendererType.Checkbox &&
+        columnConfig.visible
```

Comment:
Why only filter visible columns?

Reply:
because, I assumed that visible columns would be the data fields, otherwise, why would they be invisible?

And, I needed this for adding a single selectable column state, but excluding it from the data column configs

Reply:
OK I looked where it was used, makes sense. Maybe worth a rename though (since these are still data configs)

Reply:
rename of?

Reply:
this function - a hidden column config is still a data config. Instead of `removeNonDataConfigs`, it could instead be `fetchableColumnConfigs` or the like.